include/arch/arm/cortex_m/scripts/linker.ld

```diff
@@ -163,6 +163,16 @@ SECTIONS
 
 	_image_text_end = .;
 
+#if defined (__GCC_LINKER_CMD__)
```

Comment:
Can we condition this on CONFIG_CPLUSPLUS instead, so that we don't include this section for pure C builds?

Reply:
For C build, the section remains empty. I think is not needed to add the CONFIG_CPLUSPLUS condition, but I can do it if you want.

Reply:
On an unrelated note. Isn't __GCC_LINKER_CMD__ always defined when linker.ld is processed?

So the ifdef can be safely removed.